File: src/binarySearch.js
const searchN = (list, n) =>
    binarySearch(list, 0, list.length - 1, n);



const binarySearch = (list, start, end, n) => {
    let m = start + Math.floor((end - start) / 2);
    if (start > end) return -1;
    return list[m] == n
        ? m
        : list[m] < n
            ? binarySearch(list, m + 1, end, n)
            : binarySearch(list, start, m - 1, n)
}


console.log(searchN([1, 2, 3, 5, 6, 9, 12, 15], 56));

File: src/binary_search.py
from typing import List


def binary_search(nums: List[int], n: int, start: int = 0, end: int = None) -> int:
    if end is None:
        end = len(nums)
        
    if start >= end:
        return -1
    
    mid = (start + end) // 2
    
    if nums[mid] == n:
        return mid
    
    elif nums[mid] > n:
        return binary_search(nums, n, start, mid)
    
    else:
        return binary_search(nums, n, mid + 1, end)